PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,438 @@
+        callEnviron |= set(indexDisplay)
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        spacing = self.CELL_SIZE * 3 // 4
+        padding = 10
+        outputBox = self.canvas.create_rectangle(
+            (canvasDimensions[0] - len(self.list) * spacing - padding) // 2,
+            canvasDimensions[1] - self.CELL_SIZE - padding,
+            (canvasDimensions[0] + len(self.list) * spacing + padding) // 2,
+            canvasDimensions[1] - padding,
+            fill=self.OPERATIONS_BG)
+        callEnviron.add(outputBox)
+
+        self.wait(0.3)
+
+        # calculate where the value will need to move to
+        outputBoxCoords = self.canvas.coords(outputBox)
+        midOutputBox = (outputBoxCoords[3] + outputBoxCoords[1]) // 2
+
+        # create the value to move to output box
+        valueOutput = self.copyCanvasItem(self.list[len(self.list)-1].display_val)
+        valueList = (valueOutput,)
+        callEnviron.add(valueOutput)
+
+        # move value to output box
+        toPositions = (outputBoxCoords[0] + padding / 2 + (1 / 2) * spacing,
+                       midOutputBox)
+        self.moveItemsTo(valueList, (toPositions,), sleepTime=.02)
+
+        # make the value 25% smaller
+        newSize = (self.VALUE_FONT[0], int(self.VALUE_FONT[1] * .75))
+        self.canvas.itemconfig(valueOutput, font=newSize)
+
+        self.wait(0.3)
+        self.cleanUp(callEnviron)
+
+    def assignElement(
+            self, fromIndex, toIndex, callEnviron,
+            steps=CELL_SIZE // 2, sleepTime=0.01):
+        fromDrawable = self.list[fromIndex]
+
+        # get positions of "to" cell in array
+        toPositions = (self.cellCoords(toIndex), self.cellCenter(toIndex))
+
+        # create new display objects as copies of the "from" cell and value
+        newCell = self.copyCanvasItem(fromDrawable.display_shape)
+        newCellVal = self.copyCanvasItem(fromDrawable.display_val)
+        callEnviron |= set([newCell, newCellVal])
+
+        # Move copies to the desired location
+        self.moveItemsTo((newCell, newCellVal), toPositions, steps=steps,
+                         sleepTime=sleepTime)
+
+        # delete the original "to" display value and the new display shape
+        self.canvas.delete(self.list[toIndex].display_val)
+        self.canvas.delete(self.list[toIndex].display_shape)
+
+        # update value and display value in "to" position in the list
+        self.list[toIndex].display_val = newCellVal
+        self.list[toIndex].val = self.list[fromIndex].val
+        self.list[toIndex].display_shape = newCell
+        self.list[toIndex].color = self.list[fromIndex].color
+        callEnviron ^= set([newCell, newCellVal])
+        
+        # delete the original "from" display value and the new display shape
+        self.canvas.delete(self.list[fromIndex].display_val)
+        self.canvas.delete(self.list[fromIndex].display_shape)        
```

Comment:
This section correctly illustrates creating the `newCell` and `newCellVal`, adding it to the `callEnviron` set to mark them as temporary, moving them to cover the items at the `toIndex`, removing the existing items at the `toIndex`, placing the `newCell` and `newCellVal` in the `toIndex` cell, and then marking them as no longer temporary bytaking them out of the `callEnviron`.

Lines 173 and 174, however, should be removed.  The textbook code does not erase the values at the `fromIndex`.  The value in that cell remains until it is overwritten by something else.  Deleting the corresponding canvas items shows a "hole" that doesn't match what the algorithm does.